server/service/integration_core_test.go

```diff
@@ -54,6 +55,27 @@ func TestIntegrations(t *testing.T) {
 	suite.Run(t, testingSuite)
 }
 
+type slowReader struct{}
+
+func (s *slowReader) Read(p []byte) (n int, err error) {
+	time.Sleep(35 * time.Second)
```

Comment:
Let's not add a 35s sleep in our testing suite, maybe make the timeout configurable and pass 2 seconds, sleep 3?

Reply:
Agreed. I'll see if I can do that reasonably.